server-base/src/main/java/org/apache/kylin/rest/metrics/QueryMetricsFacade.java

```diff
@@ -84,6 +84,9 @@ private static void updateMetricsToLocal(SQLRequest sqlRequest, SQLResponse sqlR
         update(getQueryMetrics(projectName), sqlResponse);
 
         String cube = sqlResponse.getCube();
+        if (cube == null) {
```

Comment:
It's better use `StringUtils.isEmpty(cube)` to check.

Reply:
updated.